File: main/static/main/js/calculator.js
class CaloriesCalculator {
  constructor(weight, height, age, gender = 'm', pal, goal) {
    this.weight = weight;
    this.height = height;
    this.age = age;
    this.gender = gender;
    this.pal = pal;
    this.goal = goal;
  }
  bmr() {
    let genderConv = this.gender.toLowerCase() === 'm' ? 5 : -161;
    let rawBmr = this.weight * 10 + this.height * 6.25 - this.age * 5;
    return rawBmr + genderConv;
  }
  daily_energy() {
    var palArray = [1.2, 1.375, 1.55, 1.725, 1.9];
    return this.bmr() * palArray[this.pal];
  }
  daily_target() {
    var goalArray = [-500, 0, 500];
    return Math.round(this.daily_energy() + goalArray[this.goal]);
  }
}

document
  .getElementById('calorieCalculator')
  .addEventListener('change', (event) => {
    var weight = document.getElementById('weight').value;
    var height = document.getElementById('height').value;
    var age = document.getElementById('age').value;
    var gender = document.getElementById('gender').checked === true ? 'f' : 'm';
    var pal = document.getElementById('pal').value;
    var goal = document.getElementById('goal').value;

    var newUser = new CaloriesCalculator(weight, height, age, gender, pal, goal);
    var result = newUser.daily_target();
    var targetEl = document.getElementById('calorieOutput')

    if (!result) {
        targetEl.innerHTML = 'Information Missing';
        targetEl.classList.remove("green");
        targetEl.classList.add("red");
    } else {
        targetEl.innerHTML = newUser.daily_target() + ' Calories';
        targetEl.classList.remove("red");
        targetEl.classList.add("green");
    }
  });

File: calculator/calculator.py

class Calculator():
    """
    Calculator based on the Harris-Benedict Calculator
    """

    def __init__(self, user_form):
        WEIGHT_ADJUST = 0.453592 if user_form.get('weight_unit') == 'pound' else 1
        HEIGHT_ADJUST = 30.48 if user_form.get('height_unit') == 'feet' else 1

        self.weight = float(user_form.get('weight')) * WEIGHT_ADJUST
        self.height = float(user_form.get('height')) * HEIGHT_ADJUST
        self.age = int(user_form.get('age'))
        self.gender = 'f' if user_form.get('gender', 'm')[0] == 'f' else 'm'
        self.pal = int(user_form.get('pal', 0))
        self.goal = int(user_form.get('goal', 0))
    
    def results(self):
        palArray = [1.2, 1.375, 1.55, 1.725, 1.9]
        goalArray = [-500, 0, 500]
        bmr = int((self.weight * 10 + self.height * 6.25 - self.age * 5) + (5 if self.gender == 'm' else -161))
        daily_target = int(bmr * palArray[self.pal] + goalArray[self.goal])
        protein = int((daily_target * 0.30) / 4)
        return {'bmr': bmr, 'protein': protein, 'daily_target': daily_target}
